### ISSUE TITLE
Can't guarantee upper bounds

### ISSUE THREAD

**Issue by mstaehely** [kelloggm/checker-framework#91]
There are several instances of code similar to the following in joda-time, and do not see a way to usefully type their upper bounds.

```java
public int[] getArray(ArrayWrapperInterface wrap) {
    int size = wrap.size();
    int[] values = new int[size];
    for(int i = 0; i < size; i++) {
        values[i] = wrap.get(i);
    }
    return values;
}
```

The issue is that 'wrap' may have different sizes at runtime, as several classes implement it and use arrays of different length. The length of those arrays is given by the return value of the size method, which the index checker doesn't seem to support referencing.

**Comment by mernst** [comment]
There are two potential issues here:
 * Lists and list-like objects are not handled by the Index Checker yet.  So you should ignore the right-hand side of the assignment until the Index Checker is enhanced.
 * The left-hand side of the assignment seems to work for me.  I tried with this test case.

```java
import org.checkerframework.checker.index.qual.*;

class SizeVsLength {

  public int[] getArray(@NonNegative int size) {
    int[] values = new int[size];
    for(int i = 0; i < size; i++) {
        values[i] = 22;
    }
    return values;
  }

}
```

**Comment by kelloggm** [comment]
Closed by #1351, because the test passes.
